frontend/src/app/services/events.ts

```diff
@@ -49,19 +62,40 @@ export class EventHandler {
     stabilityCost: 0,
     vaccinationPerDay: 0,
   };
-  triggerStateHistory: Record<string, TriggerState[]> = {};
+  eventStateHistory: Record<string, EventState> = {};
 
-  evaluateDay(prevDate: string, currentDate: string, dayState: DayState) {
-    let prevState = this.triggerStateHistory[prevDate];
+  evaluateDay(prevDate: string, currentDate: string, dayState: DayState, eventMitigations: EventMitigation[]) {
+    let prevState = this.eventStateHistory[prevDate];
     if (!prevState) {
-      prevState = eventTriggers.map(et => ({trigger: et, timeout: 0}));
+      let autoid = 0;
+      const triggerStates: Record<string, TriggerState> = {};
+      eventTriggers.forEach(et => {
+        let id = `autoid-${autoid++}`;
+        if (et.id) id = et.id;
+        triggerStates[id] = {trigger: et, timeout: 0};
+      });
+      prevState = {triggerStates, data: {}};
     }
 
-    const currentState = prevState.map(et => ({...et, timeout: Math.max(0, et.timeout - 1)}));
-    this.triggerStateHistory[currentDate] = currentState;
+    const currentState: EventState = {triggerStates: {}, data: cloneDeep(prevState.data)};
```

Comment:
Why do you clone the data here?
Would not it be enough to take the dateState as it is and extend it with what you need as `{...dayState, eventMitigations}`, e.g.?

Reply:
Data can be changed by eventStateUpdate and I need to save its state for potential rewinds (if we implement them later)

> Would not it be enough to take the dateState as it is and extend it with what you need as {...dayState, eventMitigations}, e.g.?
This can be done though as you remember I was always pushing against exposing anything else than stats from the simulation state. E.g. right now, there will be a petty massive change in the internal state that will increase the total number of covid cases 4x (though only 25% are going to be detected)

Reply:
After a conversation we decided to leave this as is